lib/binance/api/request.rb

```diff
@@ -26,6 +26,8 @@ def send!(api_key_type: :none, headers: {}, method: :get, path: '/', params: {},
             raise Error.new(message: "invalid http method used: #{method}")
           end
           process!(response: response || '{}')
+        rescue Net::OpenTimeout, Net::ReadTimeout, Errno::ECONNRESET => error
+          raise Error.new(message: "Binance connection error: #{error.class.to_s} - #{error.to_s}")
```

Comment:
You could also add the `cause: error` bit so we have all the details of the original exception.

Reply:
Will do!